nbgitpuller/handlers.py

```diff
@@ -38,6 +48,46 @@ def emit(self, data):
         self.write('data: {}\n\n'.format(serialized_data))
         yield self.flush()
 
+    def setup_plugins(self, provider):
+        pm = pluggy.PluginManager("nbgitpuller")
+        pm.add_hookspecs(hookspecs)
+        num_loaded =pm.load_setuptools_entrypoints("nbgitpuller", name=provider)
+        if num_loaded == 0:
+            raise ProviderException(f"The provider key you supplied in the URL could not be found: {provider}")
+        return pm
+
+    @gen.coroutine
+    def progress_loop(self, queue):
```

Comment:
I'd like to make this named something more obvious, like `wait_for_sync` or `wait_for_sync_progress_queue` or similar, perhaps also with a leading `_` to clarify its an internal function not relevant for anyone else to mess around with.

When we call `progress_loop` it is quite hard for me to understand what it means, or that its blocking - which is very critical to understand I think.